neard/src/config.rs

```diff
@@ -694,6 +697,19 @@ pub fn init_configs(
             config.telemetry.endpoints.push(MAINNET_TELEMETRY_URL.to_string());
             config.write_to_file(&dir.join(CONFIG_FILENAME));
 
+            if let Some(url) = download_genesis_url {
+                download_genesis(
+                    &url.to_string(),
+                    &genesis.expect("Must specify genesis file path to download").to_string(),
+                );
+            } else if download {
+                let url = get_genesis_url(&chain_id);
+                download_genesis(
+                    &url,
+                    &genesis.expect("Must specify genesis file path to download").to_string(),
+                );
```

Comment:
There seems to be some unnecessary repetition here. You can do something like
```
let url = download_genesis_url.unwrap_or_else(|| get_genesis_url(&chain_id))
```

Reply:
Hmm, one is of str type and the other of String, as well we still need to check download is true or download-genesis-url is empty so the code is still long.

Reply:
I see.